brion/plugin/compartmentReportBinary.cpp

```diff
@@ -641,9 +676,15 @@ bool CompartmentReportBinary::_parseMapping()
     uint64_t dataOffset = get<uint64_t>(ptr, DATA_INFO + offset);
     if (_header.byteswap)
         lunchbox::byteswap(dataOffset);
-    _header.dataBlockOffset = dataOffset;
 
-    std::unique_ptr<uint8_t[]> buffer(new uint8_t[_header.dataBlockOffset]);
+    if (_ioAPI == IOapi::posix_aio)
+    {
+        if (!_remapFile(dataOffset))
+            return false;
+        ptr = reinterpret_cast<const uint8_t*>(_file.data());
+    }
+
+    std::unique_ptr<uint8_t[]> buffer(new uint8_t[dataOffset]);
```

Comment:
You have removed the assignment to _header.dataBlockOffset but in fact it's line 686 in the original code what looks redundant. Another refactoring I'd do is to move dataBlockOffset to a separate member variable of CompartmentReportBinary because in fact it's not part of the header.
Assuming that the first cell in the report is at the beginning of the data block sounds like a potential bug to me. Let's keep the assumption for the moment, but I'll try to double check with Jim when I see him.

Reply:
If you are not sure, I would prefer to not touch this now and do a proper refactoring in a separate PR.

Reply:
I'm not sure about the first cell thing. All the other comments still apply.

Reply:
done